### DISCUSSION TITLE
Update multiple instances

### DISCUSSION THREAD

**Discussion by rodolphonetto** [adonisjs/core#1624] [Help] [answered]
Hey guys, when I need to update a lot instances I'm using a loop but I dont think that this is the best way, look at the code below, when I use the .toJSON() to iterate the itens loses the .merge so I need to find it again, there is a better way for doing that?

```
    const approversToCleanUnity = await User.query()
      .where('unity_id', id)
      .fetch()

    for (const approver of approversToCleanUnity.toJSON()) {
      const approverToClean = await User.findOrFail(approver.id)

      approverToClean.merge({ unity_id: null })
      await approverToClean.save()
    }
```

**Comment by thetutlage** [comment, marked as answer]
Why you are using `toJSON`? The main purpose of toJSON is to serialize the model to a plain Javascript object. Simply loop as follows:

```ts
for (const approver of approversToCleanUnity.rows) {
  approver.merge({ unity_id: null })
  await approver.save()
}
```

**Reply by rodolphonetto** [reply]
I Sid boy know about .rows, my bad

Thank you!
